portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -59,6 +57,29 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.sendRedirect("/index.html");
   }
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("pastComments").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comments> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String c = (String) entity.getProperty("Text");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Comments comment = new Comments(id, c, timestamp);
+      comments.add(comment);
+    }
+
```

Comment:
The above block that reads from datastore, can be moved to a readComments method

private List<Comments> readComments() {
}

Reply:
Done